src/test/java/uk/nhs/digital/nhsconnect/nhais/translator/AmendmentNameToEdifactTranslatorTest.java

```diff
@@ -0,0 +1,182 @@
+            .setOp(AmendmentPatchOperation.REMOVE)));
+
+        assertThatThrownBy(() -> translator.map(amendmentBody))
+            .isInstanceOf(FhirValidationException.class)
+            .hasMessage("Removing surnames is illegal");
+    }
+
+    @ParameterizedTest
+    @MethodSource(value = "getAddOrReplaceEnums")
+    void whenRemovingAllForenamesAndModifyingAtTheSameTime_expectException(AmendmentPatchOperation operation, SoftAssertions softly) {
+        Stream.<Supplier<Optional<AmendmentPatch>>>of(
+            jsonPatches::getFirstForename,
+            jsonPatches::getSecondForename,
+            jsonPatches::getOtherForenames)
+            .forEach(patchSupplier -> {
+                reset(jsonPatches);
+                when(patchSupplier.get()).thenReturn(Optional.of(new AmendmentPatch()
+                    .setValue(AmendmentValue.from("some_value"))
+                    .setOp(operation)));
+                when(jsonPatches.getAllForenamesPath()).thenReturn(Optional.of(new AmendmentPatch()
+                    .setOp(operation)));
+
+                softly.assertThatThrownBy(() -> translator.map(amendmentBody))
+                    .isInstanceOf(FhirValidationException.class)
+                    .hasMessage("Illegal to modify forenames and remove all at the same time");
+            });
+    }
+
+    @Test
+    void whenRemovingAnyForename_expectException(SoftAssertions softly) {
+        Stream.<Supplier<Optional<AmendmentPatch>>>of(
+            jsonPatches::getFirstForename,
+            jsonPatches::getSecondForename,
+            jsonPatches::getOtherForenames)
+            .forEach(patchSupplier -> {
+                reset(jsonPatches);
+                when(patchSupplier.get()).thenReturn(Optional.of(new AmendmentPatch()
+                    .setOp(AmendmentPatchOperation.REMOVE)
+                    .setPath("/some/json/path/")));
+
+                softly.assertThatThrownBy(() -> translator.map(amendmentBody))
+                    .isInstanceOf(FhirValidationException.class)
+                    .hasMessage("Removing /some/json/path/ is illegal. Use /name/0/given to remove all forenames instead");
+            });
+    }
+
+    @ParameterizedTest
+    @MethodSource(value = "getAddOrReplaceEnums")
+    void whenAddOrReplaceValuesAreEmpty_expectException(AmendmentPatchOperation operation) {
+        when(jsonPatches.getTitle()).thenReturn(Optional.of(new AmendmentPatch()
+            .setOp(operation)
+            .setPath("/title/")
+            .setValue(AmendmentValue.from(StringUtils.EMPTY))));
+        when(jsonPatches.getSurname()).thenReturn(Optional.of(new AmendmentPatch()
+            .setOp(operation)
+            .setPath("/surname/")
+            .setValue(AmendmentValue.from(StringUtils.EMPTY))));
+        when(jsonPatches.getFirstForename()).thenReturn(Optional.of(new AmendmentPatch()
+            .setOp(operation)
+            .setPath("/first_forename/")
```

Comment:
These first_forename, second_forename, other_forename aren't valid patch paths. I don't understand what is being tested.

Reply:
From this single unit test perspective it doesn't matter how json paths look like. This test asserts that whatever is set as `path` will end up in the exception message. 
But I agree this might be misleading so I'll change them to valid paths to not add confusion